lib/src/components/text_input/text_input.dart

```diff
@@ -0,0 +1,446 @@
+  void initState() {
+    _controller = widget.controller ?? TextEditingController();
+    _errorText = widget.errorText;
+
+    if (widget.initialValue != null) {
+      _controller.text = widget.initialValue!;
+    }
+    super.initState();
+  }
+
+  @override
+  void didUpdateWidget(covariant ZetaTextInput oldWidget) {
+    if (oldWidget.errorText != widget.errorText) {
+      setState(() {
+        _errorText = widget.errorText;
+      });
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  bool validate() => _key.currentState?.validate() ?? false;
+
+  @override
+  void reset() {
+    _key.currentState?.reset();
+    _controller.clear();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        if (widget.label != null) ...[
+          _Label(
+            label: widget.label!,
+            requirementLevel: widget.requirementLevel,
+            disabled: widget.disabled,
+          ),
+          const SizedBox(height: ZetaSpacing.x1),
+        ],
+        MouseRegion(
+          onEnter: !widget.disabled
+              ? (_) => setState(() {
+                    _hovered = true;
+                  })
+              : null,
+          onExit: !widget.disabled
+              ? (_) => setState(() {
+                    _hovered = false;
+                  })
+              : null,
+          child: TextFormField(
+            enabled: !widget.disabled,
+            key: _key,
+            controller: _controller,
+            inputFormatters: widget.inputFormatters,
+            validator: (val) {
+              setState(() {
+                _errorText = widget.validator?.call(val);
+              });
+              return _errorText;
+            },
+            textAlignVertical: TextAlignVertical.center,
+            onChanged: widget.onChange,
+            style: _baseTextStyle,
+            cursorErrorColor: _colors.error,
+            decoration: InputDecoration(
+              isDense: true,
+              contentPadding: _contentPadding,
+              filled: true,
+              prefixIcon: _prefix,
+              prefixIconConstraints: widget.prefixText != null ? _affixConstraints : null,
+              suffixIcon: _suffix,
+              suffixIconConstraints: _affixConstraints,
+              focusColor: _backgroundColor,
+              hoverColor: _backgroundColor,
+              fillColor: _backgroundColor,
+              enabledBorder: _baseBorder,
+              disabledBorder: _baseBorder,
+              focusedBorder: _focusedBorder,
+              focusedErrorBorder: _errorBorder,
+              errorBorder: widget.disabled ? _baseBorder : _errorBorder,
+              hintText: widget.placeholder,
+              errorText: _errorText,
+              hintStyle: _baseTextStyle,
+              errorStyle: const TextStyle(height: 0.001, color: Colors.transparent),
```

Comment:
interesting...
could you use 0 to fully hide it?

Reply:
Weirdly that wasn't actually hiding the text in some cases, found this as a workaround